pkg/rc/fields/fields.go

```diff
@@ -12,10 +14,64 @@ func (id ID) String() string {
 }
 
 type RC struct {
-	Id              ID
+	ID              ID
 	Manifest        pods.Manifest
 	NodeSelector    labels.Selector
 	PodLabels       labels.Set
 	ReplicasDesired int
 	Disabled        bool
 }
+
+// internal struct that simplifies JSON translation
+type jsonRC struct {
+	ID              ID
+	Manifest        []byte
+	NodeSelector    string
+	PodLabels       labels.Set
+	ReplicasDesired int
+	Disabled        bool
+}
+
+// implements encoding/json.{Marshaler,Unmarshaler}
+// Unmarshaler is especially problematic because the pods.Manifest API is
```

Comment:
if you implement `UnmarshalJSON` on `pods.Manifest`, you can use that method to mutate the object. I think it feels kinda icky to put that there since it was intentionally designed to be immutable through that interface. At the same time, this is also gross. wasn't really sure which option was grosser